consensus/hotstuff/block_producer.go

```diff
@@ -5,11 +5,12 @@ import (
 	"github.com/onflow/flow-go/model/flow"
 )
 
-// BlockProducer builds a new block proposal by building a new block payload with the the builder module,
+// BlockProducer builds a new block proposal by building a new block payload with the builder module,
 // and uses VoteCollectorFactory to create a disposable VoteCollector for producing the proposal vote.
 // BlockProducer assembles the new block proposal using the block payload, block header and the proposal vote.
 type BlockProducer interface {
-	// MakeBlockProposal builds a new HotStuff block proposal using the given view and
-	// the given quorum certificate for its parent.
-	MakeBlockProposal(qc *flow.QuorumCertificate, view uint64) (*model.Proposal, error)
+	// MakeBlockProposal builds a new HotStuff block proposal using the given view,
+	// the given quorum certificate for its parent and [maybe] timeout certificate for last view(could be nil).
+	// No errors are expected during normal operation.
+	MakeBlockProposal(qc *flow.QuorumCertificate, view uint64, lastViewTC *flow.TimeoutCertificate) (*model.Proposal, error)
```

Comment:
since we are already changing the signature here, could we oder the arguments differently:
1. view
2. qc
3. lastViewTC

To me, this would be more intuitive, because the view is the most defining property of the new block, so I feel it should go first.